genesyscloud/outbound_contactlistfilter/genesyscloud_outbound_contactlistfilter_proxy.go

```diff
@@ -0,0 +1,193 @@
+		if err != nil {
+			return nil, fmt.Errorf("failed to get page of contact list filter: %v", err)
+		}
+
+		if contactListFilters.Entities == nil || len(*contactListFilters.Entities) == 0 {
+			break
+		}
+
+		for _, contactListFilter := range *contactListFilters.Entities {
+			allContactlistfilters = append(allContactlistfilters, contactListFilter)
+		}
+	}
+
+	return &allContactlistfilters, nil
+}
+
+// getOutboundContactlistfilterIdByNameFn is an implementation of the function to get a Genesys Cloud outbound contactlistfilter by name
+func getOutboundContactlistfilterIdByNameFn(ctx context.Context, p *outboundContactlistfilterProxy, name string) (id string, retryable bool, err error) {
+	contactListFilters, err := getAllOutboundContactlistfilterFn(ctx, p)
+	if err != nil {
+		return "", true, fmt.Errorf("error searching outbound contact list filter %s: %s", name, err)
+	}
+
+	var filter platformclientv2.Contactlistfilter
+	for _, contactListFilter := range *contactListFilters {
+		if *contactListFilter.Name == name {
+			log.Printf("Retrieved the contact list filter id %s by name %s", *contactListFilter.Id, name)
+			filter = contactListFilter
+			return *filter.Id, false, nil
+		}
+	}
+
+	return "", true, nil
+}
+
+// getOutboundContactlistfilterByIdFn is an implementation of the function to get a Genesys Cloud outbound contactlistfilter by Id
+func getOutboundContactlistfilterByIdFn(ctx context.Context, p *outboundContactlistfilterProxy, id string) (outboundContactlistfilter *platformclientv2.Contactlistfilter, statusCode int, err error) {
+	contactListFilter, resp, err := p.outboundApi.GetOutboundContactlistfilter(id)
+	if err != nil {
+		return nil, resp.StatusCode, err
+	}
+
+	return contactListFilter, resp.StatusCode, nil
+}
+
+// updateOutboundContactlistfilterFn is an implementation of the function to update a Genesys Cloud outbound contactlistfilter
+func updateOutboundContactlistfilterFn(ctx context.Context, p *outboundContactlistfilterProxy, id string, outboundContactlistfilter *platformclientv2.Contactlistfilter) (*platformclientv2.Contactlistfilter, error) {
+	contactListFilter, _, err := p.outboundApi.GetOutboundContactlistfilter(id)
+	if err != nil {
+		return nil, err
+	}
+
+	outboundContactlistfilter.Version = contactListFilter.Version
+	outboundContactlistfilter, _, updateErr := p.outboundApi.PutOutboundContactlistfilter(id, *outboundContactlistfilter)
+	if updateErr != nil {
+		return nil, updateErr
+	}
+
+	return outboundContactlistfilter, nil
+}
+
+// deleteOutboundContactlistfilterFn is an implementation function for deleting a Genesys Cloud outbound contactlistfilter
+func deleteOutboundContactlistfilterFn(ctx context.Context, p *outboundContactlistfilterProxy, id string) (response *platformclientv2.APIResponse, err error) {
+	resp, err := p.outboundApi.DeleteOutboundContactlistfilter(id)
+	if err != nil {
+		return resp, err
+	}
+
+	return resp, nil
+}
```

Comment:
Not important, but I think you could just `return p.outboundApi.DeleteOutboundContactlistfilter(id)` directly here. It works out the same